### DISCUSSION TITLE
MSO5: how do I get cursor value for MSO5? Does it support this function?

### DISCUSSION THREAD

**Discussion by HippeLv** [tektronix/tm_devices#281] [Q&A] [answered]
I have tried the below commands, but none of them works.
cursor_value = scope.query('CURSOR:POSITION?')
cursor_value = scope.query("CURSOR:X2?")
cursor_value = scope.commands.cursor.query()

**Comment by amccann-Tek** [comment, marked as answer]
Hi @HippeLv,

This question slipped through the cracks. Hopefully you were able to resolve this, but if not:

The command group for cursors is not very intuitive. Since our drivers mirror the SCPI underneath it is similarly not intuitive. The general structure is:
`scope.commands.display.waveview1.cursor.cursor1.state.write('1')`

Cursors are a element of the view they reside within, so if you have waveform views (waveview) and spectrum views you would have to go through each of them access the specific cursors.

There are some limitations of automatic control with these types of commands, so directly writing the SCPI as you do in your initial comment might be necessary if you are getting advanced. Those SCPI commands follow the same pattern though:
`DISplay:WAVEView<x>:CURSor:CURSOR<x>:STATE`

Please let us know if you have further questions, and again, apologies for missing this.
-Andrew

**Comment by pradhanfortive** [comment]
Thanks Andrew,

For the VBAR cusor,  these are the SCPI commands and corresponding python driver calls:
 
# DISplay: WAVEView < x >:CURSor: CURSOR < x >:VBArs: APOSition
scope.commands.display.waveview1.cursor.cursor1.vbars.aposition.write(0.1)
scope.commands.display.waveview1.cursor.cursor1.vbars.aposition.query()
 
# DISplay: WAVEView < x >:CURSor: CURSOR < x >:VBArs: BPOSition
scope.commands.display.waveview1.cursor.cursor1.vbars.bposition.write(0.2)
scope.commands.display.waveview1.cursor.cursor1.vbars.bposition.query()
 
# DISplay: WAVEView < x >:CURSor: CURSOR < x >:VBArs: DELTa? (Query Only)
scope.commands.display.waveview1.cursor.cursor1.vbars.delta.query()
 
# DISplay: WAVEView < x >:CURSor: CURSOR < x >:VBArs: UNIts? (Query Only)
scope.commands.display.waveview1.cursor.cursor1.vbars.units.query()
